Adds a craftable ghost costume

Winterous opened the pull request:
Adds the ability to craft a ghost sheet from a bedsheet and wirecutters from the crafting menu, because it's fun.
Right now it's a simple article of clothing, but in future I may make it modify the way you speak, for added spook.

Hopefully I've done everything right this time.

Winterous committed:
Merge pull request #2 from tgstation/master

Update

Winterous committed:
Merge pull request #3 from tgstation/master

update

Winterous committed:
Adds ghost sheet

nicbn commented:
Please remove the craft and the separate item, make it work by just putting bedsheet on your head or exosuit slot

Winterous commented:
Why? You cut out eyeholes. Also is it even possible for a clothing item two have different sprites and properties based upon where you equip it? Seems like a lot of work for making it make less sense.

fludd12 commented:
It's very possible, yeah; make the clothing item state change depending on what slot it is equipped in. I think I'd prefer making it just depend on the item, rather than craftable, too.

Winterous commented:
Well if a bunch of people care that much I guess I'll change it, but I really don't see a reason to have it that way over this.

epochayur commented:
what's wrong with crafting it?

Winterous committed:
Update miscellaneous.dm

Winterous committed:
Update miscellaneous.dm

Winterous committed:
So uh

I didn't consider what it would look like when you hold items in your hand while wearing this. So, extremely weak justification!

Dorsisdwarf commented:
Crafting it is fine, why would anyone object.

I dont know how robust the crafting system is but it should probbably be "any SHARP item" rather than just wiresnips

fludd12 commented:
*shrugs*. If cake is desired, let them eat cake. I guess crafting is best! Personally, id go with the head equip idea, but the spritework is great either way, and I like how it turns out.

Another idea is to use the alternate appearance system to make the player look like a ghost but only to themselves while wearing the costume, like how we do with the cardborg suit

Winterous commented:
But that would be less cute looking.

Razharas commented:
Crafting was done 3(4?) years ago when tools were still just items and component system was not heard of
Technically chaning the check_tools proc to instead of types check the components would be not too difficult

Enzarel commented:
If this ends up being crafting + tie slot instead of exosuit or hat slot we can have both space ghosts and ghosts with hats

fludd12 commented:
My only concern with that, @Enzarel, is that space ghosts means concealable space suits, which could be a bit of a problem.

ExcessiveUseOfCobblestone commented:
There is an admin item that has a ghost sprite (forgot what the path is atm), perhaps you can replace that while you're at it.

Winterous commented:
Yeah this is barely more of a disguise than wearing a mask, definitely don't want it over bulky high-value suits.
And yeah the admin ghost sprite is weird, I could replace it, but it's someone else's work so I'll leave that to them.

ninjanomnom commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -594,3 +594,12 @@
+	item_state = "ghost_sheet_item"
+	w_class = WEIGHT_CLASS_TINY
+	flags_inv = HIDEGLOVES|HIDEMASK|HIDEEARS|HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
+	alternate_worn_layer = 6	//I think this should be right? Same layer as head items, so hats will display over it I think.
```
I'm not sure if same layer will render properly but you need to use a define here regardless. You'll probably want to define a UNDER_HEAD_LAYER

Winterous replied:
Ok done, but the (X_LAYER + 1) thing confuses me, would this be identical to 7 / MASK_LAYER?
Although thinking about it, it makes more sense to do it with the defines because then it's futureproof if the layer numbers change.

Winterous committed:
Add UNDER_HEAD_LAYER

Winterous committed:
Update miscellaneous.dm

Winterous commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -594,3 +594,12 @@
+	name = "ghost sheet"
+	desc = "The hands float by themselves, so it's extra spooky."
+	icon_state = "ghost_sheet"
+	item_state = "ghost_sheet_item"
```
For some reason this icon isn't displaying correctly, it's showing a series of boxes, like it's trying to display the entire .dmi at once or something. I have not the slightest idea why.

Winterous commented on code/modules/crafting/recipes.dm:
```diff
@@ -669,3 +669,11 @@
+	result = /obj/item/clothing/suit/ghost_sheet
+	time = 5
+	tools = list(TOOL_WIRECUTTER)
+	reqs = list(/obj/item/bedsheet = 1)
```
This allows you to use any kind of bedsheet, I was intending for it to only use the plain white kind, and I don't know how to restrict it to that type. It realistically doesn't matter, but if I knew how I'd change it.

Winterous commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -594,3 +594,12 @@
+	icon_state = "ghost_sheet"
+	item_state = "ghost_sheet_item"
+	w_class = WEIGHT_CLASS_TINY
+	flags_inv = HIDEGLOVES|HIDEMASK|HIDEEARS|HIDEFACE|HIDEHAIR|HIDEFACIALHAIR
```
Doesn't hide hair and facial hair as I intended, because those checks are only made for head and face items, but it's something you can work around by shaving yourself and it doesn't really matter anyway.

Winterous commented:
There's three issues I've noted and commented on the code, but the only important one is that the icon doesn't display correctly; it shows a grid of little boxes, like it's trying to display an entire .dmi file.
I have no idea why that's happening, and no idea how to fix it. If someone could advise me that'd be great.

ninjanomnom commented:
The irc is your best bet for code help
